java/src/com/ibm/streamsx/rest/StreamingAnalyticsService.java

```diff
@@ -110,34 +109,23 @@ static StreamingAnalyticsService of(JsonElement vcapServices,
     
     /**
      * Submit a Streams bundle to run on the Streaming Analytics Service.
-     * <p>The JSON object may contain an optional {@code deploy} member that
-     * includes deployment information.
      * @param bundle A streams application bundle
-     * @param submission Deployment info to be submitted.
-     * @return The job id, or -1. Results from the submit will be added to the
-     * submission parameter object as a member named @{code submissionResults}.
+     * @param jco Job configuration overlay in JSON format.
+     * @return The Job, or null.
      * @throws IOException
      */
-    BigInteger submitJob(File bundle, JsonObject submission) throws IOException;
+    Job submitJob(File bundle, JsonObject jco) throws IOException;
 
     /**
      * Submit an archive to build on the Streaming Analytics Service, and submit
      * the job if the build is successful.
-     * <p>
-     * The JSON object contains two keys:
-     * <UL>
-     * <LI>{@code deploy} - Optional - Deployment information.</LI>
-     * <LI>{@code graph} - Required - JSON representation of the topology graph.</LI>
-     * </UL>
-     * <p>
-     * Results are added to the submission parameters object.
      * @param archive The application archive to build.
-     * @param submission Topology and deployment info to be submitted.
-     * @return The job id, or -1. Results from the submit will be added to the
-     * submission parameter object as a member named @{code submissionResults}.
+     * @param jco Job configuration overlay in JSON format.
+     * @param buildName A name for the build, or null.
```

Comment:
So as coded this is not the build name since it is modified (I think) to include a random string at the end (which is the actual build name). So is this really a build name prefix?

Reply:
Correct, I'll do a final PR on Javadoc for cleanup at some point. I'm hoping not, but we may still need some changes to allow access to hosts with self-signed certs that might need some interface / doc changes as well.